# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/loader/ModuleLoadRequest.h
# Commit: c48ad530cfe2
# Full Hash: c48ad530cfe2161cd83e1fe52821d921a55fbae1
# Author: Yoshi Cheng-Hao Huang <allstars.chh@gmail.com>
# Date: 2025-09-24 18:03 +0000
# Regressor Bug: 1968890
# File Overlap Count: 3
# Overlapping Files: js/loader/ModuleLoadRequest.h, js/loader/ModuleLoadRequest.cpp, js/loader/ModuleLoaderBase.cpp
# Description:
#   Bug 1968890 - Part 5: Add ModuleLoadRequest::SetErroredLoadingImports(). r=jonco
# ==============================================================================

diff -r 42da5cac3411 -r c48ad530cfe2 js/loader/ModuleLoadRequest.h
--- a/js/loader/ModuleLoadRequest.h	Wed Sep 24 18:03:29 2025 +0000
+++ b/js/loader/ModuleLoadRequest.h	Wed Sep 24 18:03:30 2025 +0000
@@ -118,9 +118,19 @@
 
   const Kind mKind;
 
+  void SetErroredLoadingImports() {
+    MOZ_ASSERT(IsDynamicImport());
+    MOZ_ASSERT(IsFetching());
+    mErroredLoadingImports = true;
+  }
+
   // Type of module (JavaScript, JSON)
   const ModuleType mModuleType;
 
+  // A flag (for dynamic import) that indicates the module script is
+  // successfully fetched and compiled, but its dependencies are failed to load.
+  bool mErroredLoadingImports;
+
   // Pointer to the script loader, used to trigger actions when the module load
   // finishes.
   RefPtr<ModuleLoaderBase> mLoader;